src/main/java/com/example/bookhub/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,136 @@
+package com.example.bookhub.service;
+
+import com.example.bookhub.dto.order.CreateOrderRequestDto;
+import com.example.bookhub.dto.order.OrderItemWithoutPriceResponseDto;
+import com.example.bookhub.dto.order.OrderResponseDto;
+import com.example.bookhub.mapper.OrderMapper;
+import com.example.bookhub.model.CartItem;
+import com.example.bookhub.model.Order;
+import com.example.bookhub.model.OrderItem;
+import com.example.bookhub.model.ShoppingCart;
+import com.example.bookhub.model.User;
+import com.example.bookhub.model.enums.Status;
+import com.example.bookhub.repository.cartitem.CartItemRepository;
+import com.example.bookhub.repository.order.OrderItemRepository;
+import com.example.bookhub.repository.order.OrderRepository;
+import com.example.bookhub.repository.shoppingcart.ShoppingCartRepository;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartRepository cartRepository;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderResponseDto placeOrder(CreateOrderRequestDto requestDto) {
+        User currentUser = getCurrentUser();
+        ShoppingCart cart = cartRepository.findByUserId(currentUser.getId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Shopping cart not found for user: " + currentUser.getId()));
+        Set<CartItem> cartItems = cart.getCartItems();
+        if (cartItems.isEmpty()) {
+            throw new IllegalArgumentException("Cannot place an order with an empty cart");

```

Comment:
provide parameter to message
```suggestion
            throw new OrderProcessingException("Cannot place an order with an empty cart");
```